pageserver/src/tenant.rs

```diff
@@ -1841,26 +1843,26 @@ impl Tenant {
                 )
             })?;
 
-        // fsync the parent directory to ensure the directory entry is durable
-        if first_save {
-            target_config_path
-                .parent()
-                .context("Config file does not have a parent")
-                .and_then(|target_config_parent| {
-                    File::open(target_config_parent).context("Failed to open config parent")
-                })
-                .and_then(|tenant_dir| {
-                    tenant_dir
-                        .sync_all()
-                        .context("Failed to fsync config parent")
-                })
-                .with_context(|| {
-                    format!(
-                        "Failed to fsync on first save for config {}",
-                        target_config_path.display()
-                    )
-                })?;
-        }
+        // fsync the parent directory to ensure the directory entry is durable.
+        // before this was done conditionally on first_save, but these management actions are rare
```

Comment:
NIT: We have the first_save at hand anyway, not sure if this is really needed. If some script changes config in bulk then we can have many extra fsyncs out of that (we will have many fsyncs anyway because config file itself needs an fsync)

Feel free to keep as is. Maybe a comment for such concerns. We can change this at any time if we need to

Reply:
Note that the `first_save` will be `false` for attached tenants because it's not the initialization time. Perhaps this should be refactored more to get rid of it or renamed as `at_create_tenant` or removed altogether.

We could of course do an exists check before, but less exists checks the better not to add TOCTOU issues. It would be racy right away because we can have two HTTP requests at the same time.

If this were to use in future the mutex (for the race), tempfile (#3062), rename as target, fsync(parent_dir) then it would still require the fsync for the rename.

Reply:
Renamed `first_save` as `creating_tenant`.

Reply:
> Note that the first_save will be false for attached tenants because it's not the initialization time. 

Ah, I got it wrong, I was thinking about:

>  As an alternative we can always persist default config with attach.

So you're right.